src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -7163,49 +7166,51 @@ public B M(int i, int j)
-            IInvalidOperation (OperationKind.Invalid, Type: System.Int32, IsInvalid, IsImplicit) (Syntax: 'i')
-              Children(1):
-                  IParameterReferenceOperation: i (OperationKind.ParameterReference, Type: System.Int32, IsInvalid) (Syntax: 'i')
-            IInvalidOperation (OperationKind.Invalid, Type: System.Int32, IsInvalid, IsImplicit) (Syntax: 'j++')
-              Children(1):
-                  IIncrementOrDecrementOperation (Postfix) (OperationKind.Increment, Type: System.Int32, IsInvalid) (Syntax: 'j++')
-                    Target:
-                      IParameterReferenceOperation: j (OperationKind.ParameterReference, Type: System.Int32, IsInvalid) (Syntax: 'j')
-            IInvalidOperation (OperationKind.Invalid, Type: System.Int32, IsInvalid, IsImplicit) (Syntax: 'M2()')
-              Children(1):
-                  IInvocationOperation (System.Int32 B.M2()) (OperationKind.Invocation, Type: System.Int32, IsInvalid) (Syntax: 'M2()')
-                    Instance Receiver:
-                      null
-                    Arguments(0)
-            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Int32) (Syntax: 'X = 2')
-              Left:
-                IPropertyReferenceOperation: System.Int32 B.X { get; set; } (OperationKind.PropertyReference, Type: System.Int32) (Syntax: 'X')
-                  Instance Receiver:
-                    IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: B, IsImplicit) (Syntax: 'this')
-              Right:
-                ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 2) (Syntax: '2')
-        Next (Return) Block[B2]
-            IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: B, IsImplicit) (Syntax: 'this')
-            Leaving: {R1}
-}
-Block[B2] - Exit
-    Predecessors: [B1]
-    Statements (0)
-";
+            var expectedFlowGraph = """
+                Block[B0] - Entry
+                    Statements (0)
+                    Next (Regular) Block[B1]
+                        Entering: {R1}
+                .locals {R1}
+                {
+                    CaptureIds: [0]
+                    Block[B1] - Block
+                        Predecessors: [B0]
+                        Statements (5)
+                            IFlowCaptureOperation: 0 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'this')
+                              Value:
+                                IInstanceReferenceOperation (ReferenceKind: ContainingTypeInstance) (OperationKind.InstanceReference, Type: B) (Syntax: 'this')
+                            IInvalidOperation (OperationKind.Invalid, Type: ?, IsInvalid, IsImplicit) (Syntax: 'i')
+                              Children(1):
+                                  IOperation:  (OperationKind.None, Type: null, IsInvalid) (Syntax: 'i')
```

Comment:
What is this `NoneOperation` representing?

Reply:
To be honest I kinda just left it do its job after the fix in the binder. Perhaps we should consider how to treat this situation without defaulting to a none operation

Reply:
In this case, `i` is bound as a `BoundMethodGroup`, as an aftermath of the changes to the binder. In such a weird case though I don't think we should care much about the operation that is returned.

Reply:
That seems pretty wrong. I do not want to see that returned, and it doesn't feel right that this would be bound as a method group either.

Reply:
So the direction is to discard the bound result from the identifier name case if it's not a valid member and fallback to the previous behavior (binding to `i`, the local).